internal/jimm/auditlog/cleanup.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2025 Canonical.
+package auditlog
+
+import (
+	"context"
+	"time"
+
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+)
+
+// auditLogCleanupTime indicates that we poll at 9 AM.
+var auditLogCleanupTime = pollTimeOfDay{
+	Hours: 9,
+}
+
+// StartCleanup loop forever and checks daily for any logs
+// that need to be cleaned up. This method should be run
+// in a separate Go routine to avoid blocking, it will terminate
+// when the provided context is cancelled.
+func (j *auditLogManager) StartCleanup(ctx context.Context) {
+	if j.retentionPeriodInDays == 0 {
+		return
+	}
+	for {
+		select {
+		case <-time.After(calculateNextPollDuration(auditLogCleanupTime, time.Now().UTC())):
+			j.cleanup(ctx)
+		case <-ctx.Done():
+			zapctx.Debug(ctx, "exiting audit log cleanup polling")
+			return
+		}
+	}
+}
+
+// pollTimeOfDay holds the time hour, minutes and seconds to poll for cleanup.
+type pollTimeOfDay struct {
+	Hours   int
+	Minutes int
+	Seconds int
+}
+
+func (j *auditLogManager) cleanup(ctx context.Context) {
+	retentionDate := time.Now().AddDate(0, 0, -(j.retentionPeriodInDays))
```

Comment:
for testing purposes it would be nice if the auditLogManager had a "clock" filed (see https://pkg.go.dev/github.com/juju/clock@v1.1.1#Clock) so in tests you could pass in your own "test" clock, but in production it would just be the WallClock

Reply:
Here is the blog post I was talking about that mention's Juju and in Go 1.24 there will also be a neat way of testing time sensitive code.
https://devcenter.upsun.com/posts/go-124/#a-new-testingsynctest-package-for-testing-concurrent-code

Reply:
But for now I think this is sufficiently tested.